trie/zk_trie_impl.go

```diff
@@ -162,157 +176,176 @@ func (mt *ZkTrieImpl) pushLeaf(newLeaf *Node, oldLeaf *Node, lvl int,
 	}
 	// We can add newLeaf now.  We don't need to add oldLeaf because it's
 	// already in the tree.
-	_, err = mt.addNode(newLeaf)
+	mt.dirtyStorage[*newLeafHash] = newLeaf
+	newParentNodeKey := mt.newDirtyNodeKey()
+	mt.dirtyStorage[*newParentNodeKey] = newParentNode
+	return newParentNodeKey, nil
+}
+
+// Commit calculates the root for the entire trie and persist all the dirty nodes
+func (mt *ZkTrieImpl) Commit() error {
+	rootKey, err := mt.commit(mt.rootKey, nil, new(sync.Mutex))
```

Comment:
I recommend to consider the implement of trie in geth for a two-phase commiting, i.e.: "collect" all of the required updatings of node (i.e. items inside of `dirtyStorage`), calculate their node hashes parallelly, then flush them into the underlying db (parallel or sequential).

Reply:
see https://github.com/scroll-tech/zktrie/pull/20/commits/6812ab31fd1edade1d972e224f9c0224202dcbbc

Reply:
Ok. With this update we would not traversal the whole trie after `Root` being called